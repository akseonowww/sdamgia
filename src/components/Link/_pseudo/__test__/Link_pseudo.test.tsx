import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import LinkPseudo from '../'

describe('Link_pseudo', () => {
  let container: HTMLElement | null = null
  const handleClick = jest.fn(() =>
    console.log('You had clicked and saw less/more information')
  )

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  describe('renders two types:', () => {
    it('common', () => {
      act(() => {
        render(
          <LinkPseudo handleClick={handleClick}>Other tasks</LinkPseudo>,
          container
        )
      })
      expect(container && container.textContent).toBe('Other tasks')

      act(() => {
        render(
          <LinkPseudo className="Link_pseudoBlack" handleClick={handleClick}>
            Solution
          </LinkPseudo>,
          container
        )
      })
      expect(container && container.textContent).toBe('Solution')
    })

    it('wrap', () => {
      act(() => {
        render(
          <LinkPseudo
            className="Link_pseudoBlack ConstructorForm-TopicName"
            handleClick={handleClick}
            customChildren
          >
            <div className="ConstructorForm-TopicNumber">1</div>
            <div className="ConstructorForm-TopicDesc">
              <u className="Link-U Link_wrap-U Link_pseudo-U Link_pseudoBlack-U">
                Task topic description
              </u>
            </div>
          </LinkPseudo>,
          container
        )
      })
      expect(container && container.textContent).toBe('1Task topic description')
    })
  })

  it('should render a content', () => {
    act(() => {
      render(
        <LinkPseudo handleClick={handleClick}>Other tasks</LinkPseudo>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <LinkPseudo className="Link_pseudoBlack" handleClick={handleClick}>
          Solution
        </LinkPseudo>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <LinkPseudo
          className="Link_pseudoBlack ConstructorForm-TopicName"
          handleClick={handleClick}
          customChildren
        >
          <div className="ConstructorForm-TopicNumber">1</div>
          <div className="ConstructorForm-TopicDesc">
            <u className="Link-U Link_wrap-U Link_pseudo-U Link_pseudoBlack-U">
              Task topic description
            </u>
          </div>
        </LinkPseudo>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
