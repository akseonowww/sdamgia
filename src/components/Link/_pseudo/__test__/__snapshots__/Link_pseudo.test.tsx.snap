// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link_pseudo should render a content 1`] = `"<div class=\\"Link Link_pseudo\\">Other tasks</div>"`;

exports[`Link_pseudo should render a content 2`] = `"<div class=\\"Link Link_pseudo Link_pseudoBlack\\">Solution</div>"`;

exports[`Link_pseudo should render a content 3`] = `
"<div class=\\"Link Link_wrap Link_pseudo Link_pseudoBlack ConstructorForm-TopicName\\">
  <div class=\\"ConstructorForm-TopicNumber\\">1</div>
  <div class=\\"ConstructorForm-TopicDesc\\"><u class=\\"Link-U Link_wrap-U Link_pseudo-U Link_pseudoBlack-U\\">Task topic description</u></div>
</div>"
`;
